trajectory_msgs/JointTrajectory joint_trajectory
geometry_msgs/WrenchStamped wrench
cartesian_control_msgs/CartesianTolerance path_tolerance
cartesian_control_msgs/CartesianTolerance goal_tolerance
duration goal_time_tolerance

---

int32 error_code
int32 SUCCESSFUL = 0
int32 INVALID_GOAL = -1 # e.g. illegal quaternions in poses
int32 INVALID_JOINTS = -2
int32 OLD_HEADER_TIMESTAMP = -3
int32 PATH_TOLERANCE_VIOLATED = -4
int32 GOAL_TOLERANCE_VIOLATED = -5
int32 INVALID_POSTURE = -6

string error_string

---

Header header
string tcp_frame
cartesian_control_msgs/CartesianTrajectoryPoint desired
cartesian_control_msgs/CartesianTrajectoryPoint actual
cartesian_control_msgs/CartesianTrajectoryPoint error